Library/Homebrew/dependencies_helpers.rb

```diff
@@ -41,17 +26,14 @@ def recursive_includes(klass, root_dependent, includes, ignores)
     cache_key = "recursive_includes_#{includes}_#{ignores}"
 
     klass.expand(root_dependent, cache_key: cache_key) do |dependent, dep|
-      if dep.recommended?
-        klass.prune if ignores.include?("recommended?") || dependent.build.without?(dep)
-      elsif dep.optional?
-        klass.prune if includes.exclude?("optional?") && !dependent.build.with?(dep)
-      elsif dep.build? || dep.test?
-        keep = false
-        keep ||= dep.test? && includes.include?("test?") && dependent == root_dependent
-        keep ||= dep.build? && includes.include?("build?")
-        klass.prune unless keep
-      elsif dep.satisfied?
-        klass.prune if ignores.include?("satisfied?")
+      klass.prune if ignores.any? { |ignore| dep.send(ignore) }
```

Comment:
```suggestion
      klass.prune if ignores.any?(&method(:public_send))
```
I think this should work.

Reply:
Actually, maybe not. In any case, we should use `public_send` over `send`.

Reply:
Handled in https://github.com/Homebrew/brew/pull/15892/commits/b4be0f088b85949204a365b4c5dd2ec51a4cd0bc